### PR TITLE
Make the idris2 executable depend on the idris2 lib

### PR DESCRIPTION
**Pull request by melted**
Let's not keep two identical lists.

### CODE REVIEW

File: idris2.ipkg [hunk 1]
```diff
@@ -1,182 +1,6 @@
-package idris2
+package idris2app
```
Comment (gallais) [inline idris2.ipkg:1]: I would probably keep the package name defined by `idris2.ipkg` to be `idris2`
and make `idris2api.ipkg` define `idris2api` instead.

Comment (melted) [reply idris2.ipkg:1]: This package name will never be used as a dependency or really anywhere, the lib will, so it gets the `idris2` name.

Comment (buzden) [reply idris2.ipkg:1]: Maybe, then, rename `idris2api.ipkg` to `idris2lib.ipkg` and `idris.ipkg` to `idris2app.ipkg`?

Comment (melted) [reply idris2.ipkg:1]: That's probably better, but I want to keep these changes unobtrusive.

**Pull request merged**